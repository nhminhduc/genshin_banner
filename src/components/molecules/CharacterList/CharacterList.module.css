.characterImageList {
  background-repeat: "no-repeat";
  background-size: "cover";
}

.characterImage.MuiImageListItem-img {
  object-fit: contain;
}
