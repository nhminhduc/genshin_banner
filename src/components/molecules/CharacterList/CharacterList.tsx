import { ImageList, useMediaQuery } from "@mui/material";
import cx from "classnames";
import CharacterListItem from "components/atoms/CharacterListItem/CharacterListItem";
import { useCharacterContext } from "hooks/useCharacterContext";
import type { CharacterData } from "types/CharacterType";

const CharacterList = () => {
  const characterDataContext: CharacterData[] = useCharacterContext();
  const mediumSize = useMediaQuery("(min-width:660px)");
  const ImageListStyle = mediumSize
    ? { height: 640, width: 560 }
    : { height: 480, width: 360 };

  return (
    <ImageList
      className={cx("overflow-hidden")}
      cols={mediumSize ? 4 : 3}
      gap={4}
      rowHeight={156}
      sx={{ ...ImageListStyle }}
    >
      {characterDataContext.map((characterData) => {
        const { fromLastBanner, name, rarity } = characterData;
        const { noVersions } = fromLastBanner;
        return (
          <CharacterListItem
            key={name}
            name={name}
            noVersions={noVersions}
            rarity={rarity}
          />
        );
      })}
    </ImageList>
  );
};

export default CharacterList;
