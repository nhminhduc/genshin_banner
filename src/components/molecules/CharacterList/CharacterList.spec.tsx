import { render } from "@testing-library/react";
import CharacterContextProvider from "context/CharacterContext";
import { describe, expect, test } from "vitest";

import CharacterList from "./CharacterList";

describe("CharacterList test", () => {
  test("CharacterList load", () => {
    const { asFragment } = render(
      <CharacterContextProvider>
        <CharacterList />
      </CharacterContextProvider>,
    );
    expect(asFragment).toMatchSnapshot();
  });
});
