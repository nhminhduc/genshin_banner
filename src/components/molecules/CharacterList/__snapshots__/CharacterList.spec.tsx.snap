// Vitest Snapshot v1

exports[`CharacterList test > CharacterList load 1`] = `[Function]`;
